From cf60691bc870c30cf43175e6a734e0b644eab816 Mon Sep 17 00:00:00 2001
From: Brandon Li <brandonli@apache.org>
Date: Tue, 22 Jul 2014 22:33:14 +0000
Subject: [PATCH 361/791] HDFS-6731. Run "hdfs zkfc-formatZK" on a server in a
 non-namenode will cause a null pointer exception.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-2@1612718 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 64f9ab07d3aebec890d85a5a1e56019b68a56990)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
---
 .../hadoop/hdfs/tools/DFSZKFailoverController.java |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSZKFailoverController.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSZKFailoverController.java
index 8b7a995..a42b1e3 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSZKFailoverController.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSZKFailoverController.java
@@ -122,6 +122,11 @@ public static DFSZKFailoverController create(Configuration conf) {
           "HA is not enabled for this namenode.");
     }
     String nnId = HAUtil.getNameNodeId(localNNConf, nsId);
+    if (nnId == null) {
+      String msg = "Could not get the namenode ID of this node. " +
+          "You may run zkfc on the node other than namenode.";
+      throw new HadoopIllegalArgumentException(msg);
+    }
     NameNode.initializeGenericKeys(localNNConf, nsId, nnId);
     DFSUtil.setGenericConf(localNNConf, nsId, nnId, ZKFC_CONF_KEYS);
     
-- 
1.7.9.5

